### ISSUE TITLE
Thoroughly Comment Smart Contracts

### ISSUE THREAD

**Issue by damons** [near/core-contracts#110]
I believe it would be helpful if each function were thoroughly commented.  Each parameter should have clear descriptions, and significant sections of logic should have comments explaining what's happening.  

A good example for contract interfaces:  https://eips.ethereum.org/EIPS/eip-721

**Comment by evgenykuzyakov** [comment]
Good point. We definitely lacking some good standard for some contracts. While staking pool contract had a standard described in a NEP, it's currently lacking some guarantees that it should provide. E.g. not panic on account balance.
